### PR TITLE
Reload phone numbers, email addresses and members

### PR DESCRIPTION
**Pull request by amberwebb**
Reloads objects to support combined contacts UI.

### CODE REVIEW

File: src/persistence.coffee [hunk 1]
```diff
@@ -289,7 +289,9 @@ modifySDK = (sdk) ->
     (phoneNumber, callback) ->
       saveMemberPhoneNumber.call(this, phoneNumber, callback)
       .then((result) ->
-        sdk.loadMembers({id: phoneNumber.memberId}).then -> result
+        sdk.loadMembers({id: phoneNumber.memberId})
```
Comment (matthopson) [inline src/persistence.coffee]: I think you want to wrap this in `promises.when()`. The `.then -> result` is only being called after `loadContactPhoneNumbers` is finished - so you may end up with a race condition with the `loadMembers` call.

File: src/persistence.coffee [hunk 2]
```diff
@@ -905,6 +907,18 @@ modifySDK = (sdk) ->
           sdk.loadMembers({id: memberId})
           sdk.loadContacts({memberId: memberId})
           sdk.loadContactEmailAddresses({memberId: memberId})
+          sdk.loadContactPhoneNumbers({memberId: memberId})
```
Comment (matthopson) [inline src/persistence.coffee]: Same with these.

File: src/persistence.coffee [hunk 3]
```diff
@@ -905,6 +907,18 @@ modifySDK = (sdk) ->
...
+
+  wrapMethod sdk, 'saveContact',
+    (saveContact) ->
+      (contact, callback) ->
+        saveContact.call(this, contact, callback)
+        .then((result) ->
+          contactId = contact.id
+          sdk.loadMembers({id: contact.memberId})
+          sdk.loadContactEmailAddresses({contactId: contactId})
+          sdk.loadContactPhoneNumbers({contactId: contactId})
```
Comment (matthopson) [inline src/persistence.coffee]: And these. You're not returning the `result` after these finish.

**Comment by amberwebb** [comment]
@matthopson How about now?

**Pull request merged**